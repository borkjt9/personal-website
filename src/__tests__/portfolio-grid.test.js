import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-dom/test-utils';
import PortfolioGrid from '../portfolio/portfolio-grid/portfolio-grid';
import renderer from 'react-test-renderer';
import portfolioArr from '../public-objects/portfolio-arr';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({'adapter': new Adapter() });

function changePortfolioItem() {
  console.log('filler')
}

let wrapper;
let renderedPage;
beforeEach(()=> {
  wrapper = shallow(<PortfolioGrid changePortfolioItem={changePortfolioItem} />)

  renderedPage = TestUtils.renderIntoDocument(
        <PortfolioGrid changePortfolioItem={changePortfolioItem} />
      );


})
describe('PortfolioGrid', () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<PortfolioGrid
      changePortfolioItem={changePortfolioItem}
    />, div);
  });

  it("includes at least 1 portfolio item", () => {
      expect(wrapper.find("PortfolioItem").exists()).toBe(true);
  });

  it("includes the correct number of portfolio items", () => {
    expect(wrapper.find("PortfolioItem")).toHaveLength(portfolioArr.length);

  });

  it("includes all the portfolio items", () => {
    let portfolioItemNames = [];
    portfolioArr.forEach(item => {
      portfolioItemNames.push(item.name)

    })

    const renderedItems = TestUtils.scryRenderedDOMComponentsWithClass(renderedPage, 'portfolio__item__desc__title')
    renderedItems.forEach((item, i) => {
      expect(item.textContent).toEqual(portfolioItemNames[i])
    });

    // const renderedNames = wrapper.find("portfolio__item__desc__title").forEach(item => {
    //   expect(item.textContent).toEqual("Boon Investments")
    //   //console.log('item', item.find("Boon Investments").toEqual("Boon Investments"))
    // })
    // console.log('rendered names: ', renderedNames);


  });

  it('renders correctly', () => {
    const component = renderer.create(
      <PortfolioGrid
        changePortfolioItem={changePortfolioItem}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
