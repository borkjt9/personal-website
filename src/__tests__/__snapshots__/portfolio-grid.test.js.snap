// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioGrid renders correctly 1`] = `
<div
  className="portfolio-grid"
  style={
    Object {
      "marginTop": "115px",
    }
  }
>
  <div
    className="portfolio portfolio--wrapped portfolio-grid justify-content-around"
  >
    <div
      style={Object {}}
    >
      <div
        className="transition-border portfolio__item max-dimensions-is-screen portfolio-grid__portfolio__item"
        style={Object {}}
      >
        <button
          onClick={[Function]}
        >
          <img
            alt="Boon Investments"
            className="portfolio__item__image"
            sizes="(max-width: 250px) 95vw, 250px"
            src="https://johnborkowski.me/images/boon-investments-250.jpg"
            srcSet="https://johnborkowski.me/images/boon-investments-250.jpg 250w,
                      https://johnborkowski.me/images/boon-investments-500.jpg 500w,
                      https://johnborkowski.me/images/boon-investments-750.jpg 750w"
          />
          <div
            className="portfolio__item__desc"
          >
            <div
              className="portfolio__item__desc__title"
            >
              <h4
                className="is-animated"
              >
                Boon Investments
              </h4>
            </div>
            <div
              className="portfolio__item__desc__skills"
            >
              <h5
                className=""
              >
                UI/UX | Swift | Python
              </h5>
            </div>
          </div>
        </button>
      </div>
    </div>
    <div
      style={Object {}}
    >
      <div
        className="transition-border portfolio__item max-dimensions-is-screen portfolio-grid__portfolio__item"
        style={Object {}}
      >
        <button
          onClick={[Function]}
        >
          <img
            alt="RIA Portal"
            className="portfolio__item__image"
            sizes="(max-width: 250px) 95vw, 250px"
            src="https://johnborkowski.me/images/auto-oms-250.jpg"
            srcSet="https://johnborkowski.me/images/auto-oms-250.jpg 250w,
                      https://johnborkowski.me/images/auto-oms-500.jpg 500w,
                      https://johnborkowski.me/images/auto-oms-750.jpg 750w"
          />
          <div
            className="portfolio__item__desc"
          >
            <div
              className="portfolio__item__desc__title"
            >
              <h4
                className="is-animated"
              >
                RIA Portal
              </h4>
            </div>
            <div
              className="portfolio__item__desc__skills"
            >
              <h5
                className=""
              >
                UI/UX | Angular | AWS
              </h5>
            </div>
          </div>
        </button>
      </div>
    </div>
    <div
      style={Object {}}
    >
      <div
        className="transition-border portfolio__item max-dimensions-is-screen portfolio-grid__portfolio__item"
        style={Object {}}
      >
        <button
          onClick={[Function]}
        >
          <img
            alt="White Label Apps"
            className="portfolio__item__image"
            sizes="(max-width: 250px) 95vw, 250px"
            src="https://johnborkowski.me/images/oshares-250.jpg"
            srcSet="https://johnborkowski.me/images/oshares-250.jpg 250w,
                      https://johnborkowski.me/images/oshares-500.jpg 500w,
                      https://johnborkowski.me/images/oshares-750.jpg 750w"
          />
          <div
            className="portfolio__item__desc"
          >
            <div
              className="portfolio__item__desc__title"
            >
              <h4
                className="is-animated"
              >
                White Label Apps
              </h4>
            </div>
            <div
              className="portfolio__item__desc__skills"
            >
              <h5
                className=""
              >
                Swift | Partnerships
              </h5>
            </div>
          </div>
        </button>
      </div>
    </div>
    <div
      style={Object {}}
    >
      <div
        className="transition-border portfolio__item max-dimensions-is-screen portfolio-grid__portfolio__item"
        style={Object {}}
      >
        <button
          onClick={[Function]}
        >
          <img
            alt="AutoPOOL"
            className="portfolio__item__image"
            sizes="(max-width: 250px) 95vw, 250px"
            src="https://johnborkowski.me/images/autopool-250.jpg"
            srcSet="https://johnborkowski.me/images/autopool-250.jpg 250w,
                      https://johnborkowski.me/images/autopool-500.jpg 500w,
                      https://johnborkowski.me/images/autopool-750.jpg 750w"
          />
          <div
            className="portfolio__item__desc"
          >
            <div
              className="portfolio__item__desc__title"
            >
              <h4
                className="is-animated"
              >
                AutoPOOL
              </h4>
            </div>
            <div
              className="portfolio__item__desc__skills"
            >
              <h5
                className=""
              >
                UI/UX | Swift | App Engine
              </h5>
            </div>
          </div>
        </button>
      </div>
    </div>
    <div
      style={Object {}}
    >
      <div
        className="transition-border portfolio__item max-dimensions-is-screen portfolio-grid__portfolio__item"
        style={Object {}}
      >
        <button
          onClick={[Function]}
        >
          <img
            alt="Bank of America"
            className="portfolio__item__image"
            sizes="(max-width: 250px) 95vw, 250px"
            src="https://johnborkowski.me/images/bank-of-america-250.jpg"
            srcSet="https://johnborkowski.me/images/bank-of-america-250.jpg 250w,
                      https://johnborkowski.me/images/bank-of-america-500.jpg 500w,
                      https://johnborkowski.me/images/bank-of-america-750.jpg 750w"
          />
          <div
            className="portfolio__item__desc"
          >
            <div
              className="portfolio__item__desc__title"
            >
              <h4
                className="is-animated"
              >
                Bank of America
              </h4>
            </div>
            <div
              className="portfolio__item__desc__skills"
            >
              <h5
                className=""
              >
                Modeling | Forecasting
              </h5>
            </div>
          </div>
        </button>
      </div>
    </div>
    <div
      style={Object {}}
    >
      <div
        className="transition-border portfolio__item max-dimensions-is-screen portfolio-grid__portfolio__item"
        style={Object {}}
      >
        <button
          onClick={[Function]}
        >
          <img
            alt="SunRun"
            className="portfolio__item__image"
            sizes="(max-width: 250px) 95vw, 250px"
            src="https://johnborkowski.me/images/sunrun-250.jpg"
            srcSet="https://johnborkowski.me/images/sunrun-250.jpg 250w,
                      https://johnborkowski.me/images/sunrun-500.jpg 500w,
                      https://johnborkowski.me/images/sunrun-750.jpg 750w"
          />
          <div
            className="portfolio__item__desc"
          >
            <div
              className="portfolio__item__desc__title"
            >
              <h4
                className="is-animated"
              >
                SunRun
              </h4>
            </div>
            <div
              className="portfolio__item__desc__skills"
            >
              <h5
                className=""
              >
                FP&A | IPO
              </h5>
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
